docs/source/onnx_dynamo.rst

```diff
@@ -0,0 +1,151 @@
+TorchDynamo-based ONNX Exporter
+===============================
+
+.. automodule:: torch.onnx
+  :noindex:
+
+.. contents:: :local:
+    :depth: 3
+
+.. warning::
+  The ONNX exporter for TorchDynamo is under active development and is subject to rapid change.
+
+Overview
+--------
+
+As mentioned before, the ONNX exporter leverages TorchDynamo engine to hook into Python's frame evaluation API
+and dynamically rewrite its bytecode into an FX Graph.
+The resulting FX Graph is then polished before it is finally translated into an ONNX graph.
+
+The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>_` is captured using
+bytecode analysis that preserves the dynamic nature of the model instead of using traditional static tracing techniques.
+
+The exporter is designed to be modular and extensible. It is composed of the following components:
+
+  - **ONNX Exporter**: :class:`Exporter` main class that orchestrates the export process.
+  - **ONNX Export Options**: :class:`ExportOptions` has a set of options that control the export process.
+  - **ONNX Registry**: :class:`OnnxRegistry` is the registry of ONNX operators and functions.
+  - **FX Graph Extractor**: :class:`FXGraphExtractor` extracts the FX graph from the PyTorch model.
+  - **Fake Mode**: :class:`ONNXFakeContext` is a context manager that enables fake mode for large scale models.
```

Comment:
Links for these two are not working: 
![image](https://github.com/pytorch/pytorch/assets/11205048/e4378fca-f0f4-4b3f-bda0-80080df24f1d)

Reply:
Yes, that happens because they are not part of `torch/onnx/__init__.py`, so the `automodule` directive does not capture.

I've tried to find a way to link to private classes in sphinx, but didn't find anything yet.